src/main/kotlin/com/pubnub/api/retry/RetryConfiguration.kt

```diff
@@ -0,0 +1,109 @@
+     */
+    class Linear internal constructor(
+        var delayInSec: Duration = MIN_DELAY.seconds,
+        var maxRetryNumber: Int = MAX_RETRIES_IN_LINEAR,
+        val excludedOperations: List<RetryableEndpointGroup> = emptyList(),
+        isInternal: Boolean = false
+    ) : RetryConfiguration() {
+        private val log = LoggerFactory.getLogger(this.javaClass.simpleName + "-" + "RetryConfiguration")
+
+        constructor(
+            delayInSec: Int = MIN_DELAY, // min value is 2
+            maxRetryNumber: Int = MAX_RETRIES_IN_LINEAR, // max value is 10
+            excludedOperations: List<RetryableEndpointGroup> = emptyList(),
+        ) : this(delayInSec.seconds, maxRetryNumber, excludedOperations, false)
+
+        init {
+            if (!isInternal) {
+                if (delayInSec < MIN_DELAY.seconds) {
+                    log.trace("Provided delay is less than $MIN_DELAY, setting it to $MIN_DELAY")
+                    delayInSec = MIN_DELAY.seconds
+                }
+                if (maxRetryNumber > MAX_RETRIES_IN_LINEAR) {
+                    log.trace("Provided maxRetryNumber is greater than $MAX_RETRIES_IN_LINEAR, setting it to $MAX_RETRIES_IN_LINEAR")
+                    maxRetryNumber = MAX_RETRIES_IN_LINEAR
+                }
+            }
+        }
+    }
+
+    /**
+     * This class represents an exponential retry policy with a minimum and
+     * maximum delay between retries, a maximum number of retries, and a list of
+     * operations to exclude from retry attempts.
+     *
+     * @property minDelayInSec The minimum delay in seconds between retries. Minimum value is 3 seconds.
+     * @property maxDelayInSec The maximum delay in seconds between retries.
+     * @property maxRetryNumber The maximum number of retries allowed. Maximum value is 10.
+     * @property excludedOperations List of [RetryableEndpointGroup] to be excluded from retry.
+     */
+    class Exponential internal constructor(
+        var minDelayInSec: Duration = MIN_DELAY.seconds, // min value is 2
+        var maxDelayInSec: Duration = MAX_DELAY.seconds, // max value is 150
+        var maxRetryNumber: Int = MAX_RETRIES_IN_EXPONENTIAL, // max value is 6
+        val excludedOperations: List<RetryableEndpointGroup> = emptyList(),
+        isInternal: Boolean = false
+    ) : RetryConfiguration() {
+        private val log = LoggerFactory.getLogger(this.javaClass.simpleName + "-" + "RetryConfiguration")
+
+        constructor(
+            minDelayInSec: Int = MIN_DELAY, // min value is 2
+            maxDelayInSec: Int = MAX_DELAY, // max value is 150
+            maxRetryNumber: Int = MAX_RETRIES_IN_EXPONENTIAL, // max value is 6
+            excludedOperations: List<RetryableEndpointGroup> = emptyList(),
+        ) : this(minDelayInSec.seconds, maxDelayInSec.seconds, maxRetryNumber, excludedOperations, false)
+
+        init {
+            if (!isInternal) {
+                if (minDelayInSec < MIN_DELAY.seconds) {
```

Comment:
If minDelayInSec in lower that 2 e.g. 1, 0, -1 then it will be set to MIN_DELAY.
Your remark inspired me to refactor this code to something like: 

```
if (!isInternal) {
    minDelayInSec = minDelayInSec.coerceIn(MIN_DELAY.seconds, MAX_DELAY.seconds)
    maxDelayInSec = maxDelayInSec.coerceAtLeast(minDelayInSec).coerceAtMost(MAX_DELAY.seconds)
    maxRetryNumber = maxRetryNumber.coerceAtMost(MAX_RETRIES_IN_EXPONENTIAL)

    log.trace("Adjusted values: minDelayInSec=$minDelayInSec, maxDelayInSec=$maxDelayInSec, maxRetryNumber=$maxRetryNumber")
}
```